core/src/main/java/com/github/kaktushose/jda/commands/embeds/Embed.java

```diff
@@ -0,0 +1,317 @@
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private JsonNode localize(JsonNode node) {
+        if (node instanceof ObjectNode objectNode) {
+            Iterator<Map.Entry<String, JsonNode>> iterator = objectNode.fields();
+            while (iterator.hasNext()) {
+                Map.Entry<String, JsonNode> entry = iterator.next();
+                JsonNode child = entry.getValue();
+                JsonNode newChild = localize(child);
+                if (newChild.isTextual()) {
+                    objectNode.put(entry.getKey(), i18n.localize(locale, newChild.asText(), placeholders));
+                } else {
+                    objectNode.set(entry.getKey(), newChild);
+                }
+            }
+        } else if (node instanceof ArrayNode arrayNode) {
+            for (int i = 0; i < arrayNode.size(); i++) {
+                JsonNode child = arrayNode.get(i);
+                arrayNode.set(i, localize(child));
+            }
+        }
+        return node;
+    }
+
+    /// Transforms this embed into [MessageCreateData].
+    @NotNull
+    public MessageCreateData toMessageCreateData() {
+        return MessageCreateData.fromEmbeds(build());
+    }
+
+    /// Transforms this embed into [MessageEditData].
+    @NotNull
+    public MessageEditData toMessageEditData() {
+        return MessageEditData.fromEmbeds(build());
+    }
+
+    /// Methods for manipulating the fields of an [Embed].
+    public interface Fields {
+
+        /// Removes all fields of this embed based on the given [Predicate].
+        ///
+        /// @param filter the [Predicate] to test the fields with
+        /// @return this instance for fluent interface
+        @NotNull
+        Fields removeIf(@NotNull Predicate<MessageEmbed.Field> filter);
+
+        /// Removes all fields with the given name of this embed based on the given [Predicate].
+        ///
+        /// @param name   the name of the fields to test
+        /// @param filter the [Predicate] to test the fields with
+        /// @return this instance for fluent interface
+        @NotNull
+        Fields removeIf(@NotNull String name, Predicate<MessageEmbed.Field> filter);
+
+        /// Removes all fields with the given value.
+        ///
+        /// @param value the value of a field that should be removed
+        /// @return this instance for fluent interface
+        @NotNull
+        default Fields remove(@NotNull String value) {
+            return removeIf(field -> value.equals(field.getValue()));
+        }
+
+        /// Removes all fields with the given name.
+        ///
+        /// @param name the name of a field that should be removed
+        /// @return this instance for fluent interface
+        @NotNull
+        default Fields removeByName(@NotNull String name) {
+            return removeIf(field -> name.equals(field.getName()));
```

Comment:
Calling `removeIf(String, Predicate)` with the predicate returning always true would be better to avoid duplicate logic.

Reply:
wdym?